react/OrderItems.tsx

```diff
@@ -570,6 +506,96 @@ export const OrderItemsProvider: FC = ({ children }) => {
     [enqueueTask, setOrderForm, updateItemsTask]
   )
 
+  /**
+   * Add the items to the order form.
+   *
+   * Returns if the items were added or not.
+   */
+  const addItem = useCallback(
+    (
+      items: Array<Partial<CatalogItem>>,
+      marketingData?: Partial<MarketingData>
+    ) => {
+      const { newItems, updatedItems } = items.reduce(
+        (acc, item) => {
+          const { newItems: newList, updatedItems: updateList } = acc
+          const existingItem = findExistingItem(item, orderFormItemsRef.current)
+
+          if (existingItem == null) {
+            newList.push(item)
+          } else {
+            updateList.push(item)
+          }
+
+          return acc
+        },
+        { newItems: [], updatedItems: [] } as Record<
+          string,
+          Array<Partial<CatalogItem>>
+        >
```

Comment:
if you want, you could type the `reduce` call itself to avoid having this `as ...` here, like

```ts
const { newItems, updatedItems } = items.reduce<{
  newItems: Array<Partial<CatalogItem>>
  updatedItems: Array<Partial<CatalogItem>>
}>(/* ... */)
```